### PR TITLE
[extension/encodingextension]Introduce zipkin and otlp encoding extensions.

### PR DESCRIPTION
**Pull request by dao-jun**
**Description:** <Describe what has changed.>

Introduce zipkin and otlp encoding extensions.

**Link to tracking Issue:** <Issue number if applicable>

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28686

**Testing:** <Describe what testing was performed and which tests were added.>

Unit tests.

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -93,6 +93,8 @@ extension/basicauthextension/                                           @open-te
 extension/bearertokenauthextension/                                     @open-telemetry/collector-contrib-approvers @jpkrohling @pavankrish123 @frzifus
 extension/encoding/jaegerencodingextension/                             @open-telemetry/collector-contrib-approvers @MovieStoreGuy @atoulme
 extension/encoding/textencodingextension/                               @open-telemetry/collector-contrib-approvers @MovieStoreGuy @atoulme
+extension/encoding/otlpencodingextension/                               @open-telemetry/collector-contrib-approvers @dao-jun
```
Comment (MovieStoreGuy) [inline .github/CODEOWNERS:96]: This should have more code owners since it is the primary supported encoding.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -93,6 +93,8 @@ extension/basicauthextension/                                           @open-te
...
+extension/encoding/zipkinencodingextension/                             @open-telemetry/collector-contrib-approvers @dao-jun
```
Comment (MovieStoreGuy) [inline .github/CODEOWNERS:97]: Could you add me to the code owners here as well?

@astencel-sumo as well? (If they agree)

File: extension/encoding/otlpencodingextension/extension.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+func (ex *otlpExtension) UnmarshalLogs(buf []byte) (plog.Logs, error) {
+	return ex.logUnmarshaler.UnmarshalLogs(buf)
+}
+
+func (ex *otlpExtension) MarshalLogs(logs plog.Logs) ([]byte, error) {
+	return ex.logMarshaler.MarshalLogs(logs)
+}
+
+func (ex *otlpExtension) Start(_ context.Context, _ component.Host) error {
```
Comment (MovieStoreGuy) [inline extension/encoding/otlpencodingextension/extension.go:62]: I would avoid doing this on start, but rather shift it to a New method.

File: extension/encoding/zipkinencodingextension/extension.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+var _ ptrace.Marshaler = &zipkinExtension{}
+var _ ptrace.Unmarshaler = &zipkinExtension{}
+
+type zipkinExtension struct {
+	config      *Config
+	marshaler   ptrace.Marshaler
+	unmarshaler ptrace.Unmarshaler
+}
+
+func (ex *zipkinExtension) Start(ctx context.Context, host component.Host) error {
```
Comment (MovieStoreGuy) [inline extension/encoding/zipkinencodingextension/extension.go:33]: Move this to a new method instead of start.

**Review by MovieStoreGuy — COMMENTED** [review]
I would suggest using adding `config.Validate` instead of doing the validation within Start. 
It makes it easier to validate config with the collector's sub command.

**Comment by VihasMakwana** [comment]
@dao-jun 
@atoulme has a similar PR open for otlpencodingextension https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27263. 
We should keep only one PR for otlp I believe.

**Comment by atoulme** [comment]
Good call - maybe I should close mine in favor of this one though, it looks better imo.

**Comment by VihasMakwana** [comment]
Agree @atoulme.
This one also has a support for `proto`

**Comment by dmitryax** [comment]
@dao-jun thanks for the contribution. Can you please split the PR into 2 separate for zipkin and OTLP?

**Comment by dao-jun** [comment]
> @dao-jun thanks for the contribution. Can you please split the PR into 2 separate for zipkin and OTLP?

OK, I‘ll do it

**Pull request closed without merging**